/**:
  ros__parameters:
    debug: true
    show_pic: true
    enemy_color: 0 # 0-红色，1-蓝色    注意这个！！！！！
    bullet_speed: 27.0

    odom_coordinate: "odom"
    shooter_coordinate: "shooter"
    camera_coordinate: "camera"

    # omni
    camera_matrix: [2142.4253006101626, 0.0, 654.8800557555103, 0.0, 2139.740720699495, 247.26009197675802, 0.0, 0.0, 1.0]
    distortion_coefficients: [-0.04313325802537415, 0.3598873080850437, -0.011789027160352577, -0.0068734187976891474, 0.0]
    rune_camera_matrix: [2142.4253006101626, 0.0, 644.8800557555103, 0.0, 2139.740720699495, 547.26009197675802, 0.0, 0.0, 1.0]

/camera_node:
    ros__parameters:
        inner_shot_flag: false
        videoflag: false        # 是否使用视频流,要改
        video_path: "/home/phoenix/s25/ros_ws/src/videos/9.mp4"        # 视频路径，要改
        rune_exposure: 4000

/tf2_node:
    ros__parameters:
        # omni
        odom2shooter_r: 0.056
        shooter2camera_tvec: [0.03805, 0, 0.0295]
        shooter_coordinate: "shooter"
        camera_coordinate: "camera"
        odom_coordinate: "odom"

/armor_detector_node:
    ros__parameters:
        binary_threshold_red: 30
        binary_threshold_blue: 30

/armor_tracker_node:
    ros__parameters:
        tracker.tracking_thres: 5
        tracker.lost_time_thres: 0.5
        tracker.max_v_yaw: 0.6
        tracker.flytime_offset: 0.05
        max_match_yaw_diff: 2.5
        tracker.max_match_distance: 0.5
        # update_R - 测量噪声协方差矩阵
        ekf.r_xyz_factor: 0.05
        ekf.r_yaw: 0.02
        # update_Q - 过程噪声协方差矩阵
        ekf.sigma2_q_xyz: 20.0
        ekf.sigma2_q_yaw: 70.0
        ekf.sigma2_q_r: 800.0

/rune_detector_node:
    ros__parameters:
        confidence_threshold: 0.70 #神经网络置信度阈值
        model_path: "/model/2023/yolox.onnx" #这里的路径是相对路径
        # model_path: "/model/yolox_fp16.onnx"

/rune_tracker_node:
    ros__parameters:
        filter_astring_threshold: 20 #用于不可用数据的计数，当数据突变时候，则过滤一定数目的数据之后丢入 ukf
        chasedelay: 0.07
        phase_offset: 0.20 # 相位差补偿，用于补偿观测到的角速度和滤波后角速度的相位差
        std_a: 3.0 # 加速度的标准差
        std_yawdd: 2.09 # 角加速度的标准差

/shooter_node:
    ros__parameters:
        gravity: 9.781
        mode: 115 # small:115 big:98 light:108
        k_of_small: 0.01903
        k_of_big: 0.000556
        k_of_light: 0.00053
        stop_error: 0.001
        number_of_iterations: 60
        yaw_threshold: 0.03
        pitch_threshold: 0.01
